File: CoWorkHub/CoWorkHub.Model/UserRole.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class UserRole
    {
        public int UserRoleId { get; set; }
        public int UserId { get; set; }
        public int RoleId { get; set; }
        public DateTime? ModifiedAt { get; set; }
        public virtual Role Role { get; set; } = null!;
    }
}

File: CoWorkHub/UI/coworkhub_mobile/lib/models/user_role.dart
import 'package:json_annotation/json_annotation.dart';
import 'role.dart';

part 'user_role.g.dart';

@JsonSerializable()
class UserRole {
  int userRoleId;
  int userId;
  int roleId;
  DateTime? modifiedAt;
  Role role;

  UserRole({
    required this.userRoleId,
    required this.userId,
    required this.roleId,
    this.modifiedAt,
    required this.role,
  });

  factory UserRole.fromJson(Map<String, dynamic> json) =>
      _$UserRoleFromJson(json);
  Map<String, dynamic> toJson() => _$UserRoleToJson(this);
}
